### ISSUE TITLE
Pool migration

### ISSUE THREAD

**Issue by mfw78** [democratic-csi/democratic-csi#72]
Hi, just a quick question. Any suggestion on how to go about migrating a TrueNAS pool from one to another without affecting the PVCs / democratic-csi? I'm happy to accept some small downtime, and view this as the only option really to achieve it. Any directions are appreciated.

**Comment by travisghansen** [comment]
Yeah that’s pretty involved but possible.

Some questions:

- are you intending to zfs send/receive to actually move the data?
- nfs or iscsi or both?
- is it the same node/instance of truenas?

Essentially you’ll be looking at modified version of this: https://github.com/democratic-csi/democratic-csi/blob/master/contrib/freenas-provisioner-to-democratic-csi.sh

**Comment by mfw78** [comment]
I would be intending to use zfs send / receive to actually move the data. It's both NFS and iSCSI, and it will be to the same node.

Essentially I'm just wanting to replace pool. Trying to increase performance and have opted to go about setting up a striped mirror arrangement to increase performance. raidz is killing it.

**Comment by travisghansen** [comment]
Can you just send/receive the whole pool and then rename?

**Comment by mfw78** [comment]
@travisghansen Yes, I'm just going to send/receive the whole pool for the copy. That's no problem. This is the first time that I had migrated pools, so was a little confronting. I managed to stumble through it, with a little help from https://serverfault.com/questions/88638/moving-a-zfs-filesystem-from-one-pool-to-another. 

I know it's more a case for ZFS in general, it seems like a common problem that people (such as myself), initially setup their pool incorrectly, then need to migrate it. So much so, I'd expect to have found a relatively simple guide out there, but I was left wanting.
